lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -480,6 +485,173 @@ def post(self, request):
             branch = request.POST.get('repo_branch', '').strip().replace(' ', '').replace(';', '')
             self.msg += self.get_course_from_git(gitloc, branch)
 
+        elif action == 'download_course_users':
+            course_id = request.POST.get('course_id', '').strip()
```

Comment:
This course getting logic looks like something that could be extracted into a separate get_course() method. It could return course or None, so you do not need the separate course_found flag.